Database schema + s3 file upload

pandaponn opened the pull request:
Cleaned up database schema and included dummy data for testing purposes. Also included an UploadObject class to handle S3 upload operations for file transfer.

HarveyRDonnelly commented on src/main/java/entities/UploadObject.java:
```diff
@@ -0,0 +1,41 @@
+import java.nio.ByteBuffer;
+import java.util.Random;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
```
Hey Zack, great work here ! It would be helpful if you could share the link to this library and add it to the build.gradle file for compilation purposes.

**UPDATE:** My apologies, I see that you have updated the build.gradle !

HarveyRDonnelly commented on src/main/bin/db/initPostgresSchema.sql:
```diff
@@ -0,0 +1,172 @@
+
+DELETE FROM messages;
+INSERT INTO messages
+VALUES ('w2ilpab8', 'r5hdyrg2', 'b36as7qx', '5ogwodol', 'great answer!', current_timestamp),
```
This is great ! A small issue is that the root message needs to exist in the messages table for us to construct the DiscussionBoard tree. For rows in the messages table corresponding to root messages, the parent_id can be left as null (this indicates that it is a root node).

pandaponn replied:
I've left the parent_id as null to indicate the root nodes.

HarveyRDonnelly reviewed (changes requested):
This looks great @pandaponn ! The dummy data generation and schema are very well written and thorough. The consistent use of the ID format and insertion of hashed passwords will be very beneficial when testing our use cases. The UploadObject class is also well implemented. As noted, the root messages must exist in the messages table in order for us to construct those objects and so this requires a small update. The UploadObject entity could also benefit from the inclusion of JavaDoc, but this can be added at a later date under a new issue.

pandaponn commented:
@HarveyRDonnelly Updated the messages table with root messages

pandaponn commented:
Included solution_name and test_name for solutions and tests tables respectively. Changed certain integer values to float values to account for greater precision.

Removed UploadObject.java and S3 implementations in build.gradle as we are no longer using S3 for file storage.

@HarveyRDonnelly

HarveyRDonnelly reviewed (approved):
This looks great @pandaponn ! Thank you for updating your PR with the requested changes.
